src/main.rs

```diff
@@ -125,6 +140,49 @@ fn main() {
         startup_nsh
             .write_all(include_bytes!("startup.nsh"))
             .unwrap();
+
+        // Create user provided additional files
+        for files in additional_files {
+            let mut user_file = fs.root_dir();
+
+            let mut files = files.split(':');
+            // Safe to unwrap, there will never be an empty string passed as an arg
+            let outer = files.next().unwrap();
```

Comment:
```suggestion
            let outer = PathBuf::from(file.next().expect("Invalid --add-file argument"));                                                                                                                         
            let inner = PathBuf::from(file.next().expect("Invalid --add-file argument"));
```
Verify the whole argument early. Also use `PathBuf` to represent paths.

Reply:
Good call to use PathBuf, it seems I've spent far too much time recently within #![no_std] because I completely forgot it existed